pkg/server/chaosd/stress_attack.go

```diff
@@ -0,0 +1,133 @@
+// limitations under the License.
+
+package chaosd
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/pkg/bpm"
+	"github.com/google/uuid"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/shirou/gopsutil/process"
+	"go.uber.org/zap"
+
+	"github.com/chaos-mesh/chaosd/pkg/core"
+)
+
+const (
+	StressAttack = "stress attack"
+)
+
+func (s *Server) StressAttack(attack *core.StressCommand) (string, error) {
+	var err error
+	uid := uuid.New().String()
+
+	if err := s.exp.Set(context.Background(), &core.Experiment{
+		Uid:            uid,
+		Status:         core.Created,
+		Kind:           StressAttack,
+		RecoverCommand: attack.String(),
+	}); err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	stressngPid := ""
+
+	defer func() {
+		if err != nil {
+			if err := s.exp.Update(context.Background(), uid, core.Error, err.Error(), attack.String()); err != nil {
+				log.Error("failed to update experiment", zap.Error(err))
+			}
+			return
+		}
+
+		// use the stressngPid as recover command, and will kill the pid when recover
+		if err := s.exp.Update(context.Background(), uid, core.Success, "", stressngPid); err != nil {
+			log.Error("failed to update experiment", zap.Error(err))
+		}
+	}()
+
+	stressors := &v1alpha1.Stressors{}
+	if attack.Action == core.StressCPUAction {
+		stressors.CPUStressor = &v1alpha1.CPUStressor{
+			Stressor: v1alpha1.Stressor{
+				Workers: attack.Workers,
+			},
+			Load:    &attack.Load,
+			Options: attack.Options,
+		}
+	} else if attack.Action == core.StressMemAction {
+		stressors.MemoryStressor = &v1alpha1.MemoryStressor{
+			Stressor: v1alpha1.Stressor{
+				Workers: attack.Workers,
+			},
+			Options: attack.Options,
+		}
+	}
+
+	stressorsStr, err := stressors.Normalize()
+	if err != nil {
+		return "", err
+	}
+	log.Info("stressors normalize", zap.String("arguments", stressorsStr))
+
+	cmd := bpm.DefaultProcessBuilder("stress-ng", strings.Fields(stressorsStr)...).
+		Build()
+
+	backgroundProcessManager := bpm.NewBackgroundProcessManager()
+	err = backgroundProcessManager.StartProcess(cmd)
+	if err != nil {
+		return "", err
+	}
+	log.Info("Start stress-ng process successfully", zap.String("command", cmd.String()))
+
+	stressngPid = strconv.Itoa(cmd.Process.Pid)
+
+	return uid, nil
+}
+
+func (s *Server) RecoverStressAttack(uid string, stressngPid string) error {
+	pid, err := strconv.Atoi(stressngPid)
+	if err != nil {
+		return err
+	}
+
+	proc, err := process.NewProcess(int32(pid))
+	if err != nil {
+		return err
+	}
+
+	procName, err := proc.Name()
+	if err != nil {
+		return err
+	}
+
+	if !strings.Contains(procName, "stress-ng") {
+		log.Warn("the process is not stress-ng, maybe it is killed by manual")
+		return nil
+	}
+
+	if err := proc.Kill(); err != nil {
+		log.Error("the stress-ng process kill failed", zap.Error(err))
+		return err
+	}
+
+	// TODO: delete the experiment from db
```

Comment:
We need to update the state of the experiment.

Reply:
set status to `Revoked`?

Reply:
`Destroyed`

Reply:
done